server/core/runtime/policy/conftest_executor.go

```diff
@@ -0,0 +1,112 @@
+	"github.com/runatlantis/atlantis/server/events/models"
+)
+
+type sourceResolver interface {
+	Resolve(policySet valid.PolicySet) (string, error)
+}
+
+type policyFilter interface {
+	Filter(ctx context.Context, installationToken int64, repo models.Repo, prNum int, failedPolicies []valid.PolicySet) ([]valid.PolicySet, error)
+}
+
+type exec interface {
+	CombinedOutput(args []string, envs map[string]string, workdir string) (string, error)
+}
+
+const (
+	conftestScope = "conftest.policies"
+	internalError = "internal server error"
+)
+
+// ConfTestExecutor runs a versioned conftest binary with the args built from the project context.
+// Project context defines whether conftest runs a local policy set or runs a test on a remote policy set.
+type ConfTestExecutor struct {
+	SourceResolver sourceResolver
+	Exec           exec
+	PolicyFilter   policyFilter
+}
+
+// Run performs conftest policy tests against changes and fails if any policy does not pass. It also runs an all-or-nothing
+// filter that will filter out all policy failures based on the filter criteria.
+func (c *ConfTestExecutor) Run(prjCtx command.ProjectContext, executablePath, workdir string, envs map[string]string, extraArgs []string) (string, error) {
+	var policyArgs []Arg
+	var policyNames []string
+	var failedPolicies []valid.PolicySet
+	inputFile := filepath.Join(workdir, prjCtx.GetShowResultFileName())
+	scope := prjCtx.Scope.SubScope(conftestScope)
+
+	for _, policySet := range prjCtx.PolicySets.PolicySets {
+		path, err := c.SourceResolver.Resolve(policySet)
+		// Let's not fail the whole step because of a single failure. Log and fail silently
+		if err != nil {
+			prjCtx.Log.ErrorContext(prjCtx.RequestCtx, fmt.Sprintf("Error resolving policyset %s. err: %s", policySet.Name, err.Error()))
+			continue
+		}
+		policyArgs = append(policyArgs, NewPolicyArg(path))
+		policyNames = append(policyNames, policySet.Name)
+	}
+
+	args := ConftestTestCommandArgs{
+		PolicyArgs: policyArgs,
+		ExtraArgs:  extraArgs,
+		InputFile:  inputFile,
+		Command:    executablePath,
+	}
+	serializedArgs, err := args.build()
+	if err != nil {
+		prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
+		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+		return "", errors.Wrap(err, "building args")
+	}
+
+	// TODO: run each policy set separately and use each pass/failure decision to populate failedPolicies
+	cmdOutput, policyErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
+	if policyErr != nil {
+		failedPolicies = prjCtx.PolicySets.PolicySets
+	}
+
+	title := c.buildTitle(policyNames)
+	output := c.sanitizeOutput(inputFile, title+cmdOutput)
+	installationToken, ok := prjCtx.RequestCtx.Value(contextInternal.InstallationIDKey).(int64)
+	if !ok {
+		prjCtx.Log.WarnContext(prjCtx.RequestCtx, "missing installation token")
+		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+		return output, errors.New(internalError)
+	}
```

Comment:
Looks like we [already store](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/vcs/provider/github/request/handler.go#L172) the original GH event installation token in the ctx, so let's just fetch it from there rather than pass it through multiple function calls needed to build the project context.